tests/test_states.py

```diff
@@ -220,6 +247,394 @@ def test_squeezed_coherent(self):
+        A = None
+        d = np.array([0, 0, 0, 1, 0, 0])
+        k = 0
+
+        # prepare a squeezed displaced state
+        self.circuit.reset(pure=self.kwargs['pure'])
+        self.circuit.prepare_displaced_squeezed_state(a, r, phi, 0)
+
+        state = self.circuit.state()
+        mean, var = state.poly_quad_expectation(A, d, k, phi=self.qphi)
+
+        self.assertAlmostEqual(mean, self.mu[1], delta=self.tol)
+        self.assertAlmostEqual(var, self.cov[1, 1], delta=self.tol)
+
+    def test_linear_combination(self):
+        """Test that the correct result is returned for E(ax+bp)"""
+        self.logTestName()
+
+        A = None
+        d = np.array([0.4234, 0, 0, 0.1543, 0, 0])
+        k = 0
+
+        # prepare a squeezed displaced state
+        self.circuit.reset(pure=self.kwargs['pure'])
+        self.circuit.prepare_displaced_squeezed_state(a, r, phi, 0)
+
+        state = self.circuit.state()
+        mean, var = state.poly_quad_expectation(A, d, k, phi=self.qphi)
+
+        # E(2x+p) = 2E(x) + E(p)
+        mean_expected = d[0]*self.mu[0] + d[3]*self.mu[1]
+        self.assertAlmostEqual(mean, mean_expected, delta=self.tol)
+
+        # var(2x+p) = 4var(x)+var(p)+4cov(x,p)
+        var_expected = self.cov[0, 0]*d[0]**2 + self.cov[1, 1]*d[3]**2 + 2*d[0]*d[3]*self.cov[0, 1]
+        self.assertAlmostEqual(var, var_expected, delta=self.tol)
+
+    def test_x_squared(self):
+        """Test that the correct result is returned for E(x^2)"""
+        self.logTestName()
+
+        A = np.zeros([6, 6])
+        A[0, 0] = 2
+
+        d = None
+        k = 0
+
+        # prepare a squeezed displaced state
+        self.circuit.reset(pure=self.kwargs['pure'])
+        self.circuit.prepare_displaced_squeezed_state(a, r, phi, 0)
+
+        state = self.circuit.state()
+        mean, var = state.poly_quad_expectation(A, d, k, phi=self.qphi)
+
+        mean_ex, var_ex = self.sample_one_mode_normal_expectations(
+            self.mu, self.cov, lambda X, P, XP: A[0, 0]*X**2/2)
+        self.assertAlmostEqual(mean, mean_ex, delta=self.tol)
+        self.assertAlmostEqual(var, var_ex, delta=self.tol)
+
+    def test_p_squared(self):
+        """Test that the correct result is returned for E(p^2)"""
+        self.logTestName()
+
+        A = np.zeros([6, 6])
+        A[3, 3] = 2
+
+        d = None
+        k = 0
+
+        # prepare a squeezed displaced state
+        self.circuit.reset(pure=self.kwargs['pure'])
+        self.circuit.prepare_displaced_squeezed_state(a, r, phi, 0)
+
+        state = self.circuit.state()
+        mean, var = state.poly_quad_expectation(A, d, k, phi=self.qphi)
+
+        mean_ex, var_ex = self.sample_one_mode_normal_expectations(
+            self.mu, self.cov, lambda X, P, XP: A[3, 3]*P**2/2)
+        self.assertAlmostEqual(mean, mean_ex, delta=self.tol)
+        self.assertAlmostEqual(var, var_ex, delta=self.tol)
+
+    def test_xp(self):
+        """Test that the correct result is returned for E(xp)"""
```

Comment:
Since we force the polynomials to be symmetric, we shouldn't be testing non-symmetric operators

Reply:
Bit of lazy writing on my half. The expectation value of the polynomial xp in the phase space is identical/equivalent to the expectation value of the operator (xp+px) after Weyl quantization